mongo/datadog_checks/mongo/mongo.py

```diff
@@ -163,7 +172,13 @@ def _build_metric_list_to_collect(self):
 
     def check(self, _):
         try:
-            api = MongoApi(self._config, self.log)
+            self._check()
+        except Exception:
+            self._api_client = None
+
+    def _check(self):
+        try:
+            api = self.api_client
             self.log.debug("Connected!")
```

Comment:
This log line should move to the new `api_client` property.

Reply:
Even if it's only used here?

Reply:
Ah, misread 👍